subsys/bluetooth/host/gatt.c

```diff
@@ -2539,23 +2594,169 @@ int bt_gatt_notify_cb(struct bt_conn *conn,
 }
 
 #if defined(CONFIG_BT_GATT_NOTIFY_MULTIPLE)
+static int gatt_notify_multiple_valid_args(struct bt_conn *conn, uint16_t num_params,
+					   struct bt_gatt_notify_params *params)
+{
+	__ASSERT(params, "invalid parameters\n");
+	__ASSERT(params->attr, "invalid parameters\n");
+
+	CHECKIF(num_params < 2) {
+		/* Use the standard notification API when sending only one
+		 * notification.
+		 */
+		return -EINVAL;
+	}
+
+	CHECKIF(conn == NULL) {
+		/* Use the standard notification API to send to all connected
+		 * peers.
+		 */
+		return -EINVAL;
+	}
+
+	if (!atomic_test_bit(bt_dev.flags, BT_DEV_READY)) {
+		return -EAGAIN;
+	}
+
+	if (conn->state != BT_CONN_CONNECTED) {
+		return -ENOTCONN;
+	}
+
+#if defined(CONFIG_BT_GATT_ENFORCE_CHANGE_UNAWARE)
+	/* BLUETOOTH CORE SPECIFICATION Version 5.3
+	 * Vol 3, Part G 2.5.3 (page 1479):
+	 *
+	 * Except for a Handle Value indication for the Service Changed
+	 * characteristic, the server shall not send notifications and
+	 * indications to such a client until it becomes change-aware.
+	 */
+	if (!bt_gatt_change_aware(conn, false)) {
+		return -EAGAIN;
+	}
+#endif
+
+	/* This API guarantees an ATT_MULTIPLE_HANDLE_VALUE_NTF over the air. */
+	if (!gatt_cf_notify_multi(conn)) {
+		return -EOPNOTSUPP;
+	}
+
+	return 0;
+}
+
+static int gatt_notify_multiple_valid_params(struct bt_conn *conn, uint16_t num_params,
+					     struct bt_gatt_notify_params *params, size_t *total_len)
```

Comment:
Array before array count: 

```suggestion
static int gatt_notify_multiple_valid_params(struct bt_conn *conn, const struct bt_gatt_notify_params params[], uint16_t num_params, size_t *total_len)
```